packages/react-components/src/components/VideoGallery.test.tsx

```diff
@@ -0,0 +1,194 @@
+    // tiles to show
+    jest.spyOn(responsive, '_useContainerWidth').mockImplementation(() => {
+      return 500;
+    });
+
+    const root = mountVideoGalleryWithLocalParticipant({ localParticipant });
+
+    root.setProps({ layout: 'default' });
+    const remoteParticipants = Array.from({ length: 10 }, () =>
+      createRemoteParticipant({
+        videoStream: { isAvailable: true, renderElement: createVideoDivElement() }
+      })
+    );
+    root.setProps({ remoteParticipants });
+    expect(gridVideoTileCount(root)).toBe(DEFAULT_MAX_REMOTE_VIDEO_STREAMS + 1); // +1 for the local video stream
+    expect(root.find(HorizontalGallery).find(VideoTile).length).toBe(2);
+  });
+});
+
+const mountVideoGalleryWithLocalParticipant = (attrs: {
+  localParticipant: VideoGalleryLocalParticipant;
+}): ReactWrapper<VideoGalleryProps> => {
+  const { localParticipant } = attrs;
+  return mount(<VideoGallery localParticipant={localParticipant} />);
+};
+
+const tileCount = (root: ReactWrapper<VideoGalleryProps>): number => root.find(VideoTile).length;
+
+const videoTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(VideoTile).filterWhere((node) => {
+    return node.find(Persona).length === 0 && node.find(StreamMedia).length === 1;
+  }).length;
+
+const audioTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(VideoTile).filterWhere((node) => {
+    return node.find(Persona).length === 1 && node.find('video').length === 0;
+  }).length;
+
+const gridTileCount = (root: ReactWrapper<VideoGalleryProps>): number => root.find(GridLayout).find(VideoTile).length;
+
+const gridVideoTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(GridLayout).find(StreamMedia).length;
+
+const gridAudioTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(GridLayout).find(Persona).length;
+
+const createLocalParticipant = (attrs?: Partial<VideoGalleryLocalParticipant>): VideoGalleryLocalParticipant => {
+  return {
+    userId: attrs?.userId ?? `localParticipant-${Math.random()}`,
+    isMuted: attrs?.isMuted ?? false,
+    displayName: attrs?.displayName ?? 'Local Participant',
+    isScreenSharingOn: attrs?.isScreenSharingOn ?? false,
+    videoStream: {
+      id: attrs?.videoStream?.id ?? Math.random(),
+      isAvailable: attrs?.videoStream?.isAvailable ?? false,
+      isReceiving: attrs?.videoStream?.isReceiving ?? true,
+      isMirrored: attrs?.videoStream?.isMirrored ?? false,
+      renderElement: attrs?.videoStream?.renderElement ?? undefined
+    }
+  };
+};
+
+const createVideoDivElement = (): HTMLDivElement => {
+  const divElement = document.createElement('div');
+  divElement.innerHTML = '<video>VIDEO</video>';
+  return divElement;
+};
+
+const createRemoteParticipant = (attrs?: Partial<VideoGalleryRemoteParticipant>): VideoGalleryRemoteParticipant => {
+  return {
+    userId: attrs?.userId ?? `remoteParticipant-${Math.random()}`,
```

Comment:
Let's avoid randomness in tests as much as possible. I'd just change the type of `attrs` to _require_ `userId` to be set, and then set it from the tests (could be set using index in your Array constructor).